core/src/main/java/hudson/FilePath.java

```diff
@@ -2079,13 +2192,19 @@ public void renameTo(final FilePath target) throws IOException, InterruptedExcep
     	if(this.channel != target.channel) {
     		throw new IOException("renameTo target must be on the same host");
     	}
-        act(new SecureFileCallable<Void>() {
-            private static final long serialVersionUID = 1L;
-            public Void invoke(File f, VirtualChannel channel) throws IOException {
-                Files.move(fileToPath(reading(f)), fileToPath(creating(new File(target.remote))), LinkOption.NOFOLLOW_LINKS);
-                return null;
-            }
-        });
+        act(new RenameTo(target));
+    }
+    private class RenameTo extends SecureFileCallable<Void> {
+        private final FilePath target;
```

Comment:
Seems like a bad conversion. I would rather to do the `target.remore`conversion outside and use `File` as argument

Reply:
For purposes of this PR I am not touching @kohsuke’s logic. Whether `creating(new File(target.remote))` should be replaced with `target.creating(new File(target.remote))` I am not sure. If you care to propose such changes, no problem, but they should not buried in a simple extract-nested-class refactoring like this.